base/atomics.jl

```diff
@@ -377,26 +380,26 @@ for typ in atomictypes
                      %irv = extractvalue { $ilt, i1 } %irs, 0
                      %rv = bitcast $ilt %irv to $lt
                      ret $lt %rv
-                     """, $typ, Tuple{Ptr{$typ},$typ,$typ},
+                     """, $typ, Tuple{Ptr{$typ}, $typ, $typ},
                      unsafe_convert(Ptr{$typ}, x), cmp, new)
     end
 
     arithmetic_ops = [:add, :sub]
     for rmwop in [arithmetic_ops..., :xchg, :and, :nand, :or, :xor, :max, :min]
```

Comment:
@dillondaudert I've also only just now noticed that we're defining `and`, `nand`, `or, and `nor`, unfortunately none of which are defined for `Ptr` in base, as far as I can tell. :'(

It will be annoying to move those around, but maybe you want to move those into something like `logical_ops`, and not define them for Ptr, via the same logic as we use for the `arithmetic_ops`?
Sorry, that's annoying...

Reply:
I went ahead and implemented your suggestion.